tests/Support/SupportCollectionTest.php

```diff
@@ -4154,6 +4166,18 @@ public function testHasReturnsValidResults($collection)
         $this->assertFalse($data->has('baz'));
     }
 
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testDoesntHaveReturnsValidResults($collection)
+    {
+        $data = new $collection(['foo' => 'one', 'bar' => 'two', 1 => 'three']);
+        $this->assertFalse($data->doesntHave('foo'));
+        $this->assertFalse($data->doesntHave('foo', 'bar', 1));
+        $this->assertTrue($data->doesntHave('foo', 'bar', 1, 'baz'));
+        $this->assertTrue($data->doesntHave('baz'));
+    }
```

Comment:
I was wondering about that too. However, I've done it because I saw that the `has` method has the same double test here: [Link](https://github.com/laravel/framework/blob/master/tests/Support/SupportCollectionTest.php#L4160).

Reply:
Furthermore the `testHasReturnsValidResults()` test was added on 05.08.2019 which is newer than the `testHas()` test, so I thought that the test is important and maybe covers some newer edge case.